test/distributed/test_data_parallel.py

```diff
@@ -18,11 +24,11 @@ def __init__(self, t):
             def forward(self, x):
                 return x * self.t_rg + self.t_not_rg
 
-        m = TestModule(torch.randn(100, device='cuda', requires_grad=True))
+        m = TestModule(torch.randn(100, device='cuda', requires_grad=True, dtype=torch.double))
```

Comment:
Why did you have to add `dtype=double` here?

Reply:
This is probably related to the fact that we globally change the default precision in the tests. The precision is set like this in test files:

```
torch.set_default_dtype(torch.double)
```

It would be good to understand if this was somehow set prior to the move here (though I don't see how). You could check this by running `torch.get_default_dtype()` at this site and seeing what the dtype is.

Reply:
It would have been good to follow @ezyang's suggestion here and do some digging (or just set the default dtype).

As it is now, there is a mixture of: 1) functions that set the dtype explicitly to double (I'm guessing these are ones you changed because the precision issues were causing failures, 2) tests that don't set a dtype (I'm guessing these are meant to be double but don't happen to cause precision issues), 3) tests that set dtype to float explicitly (these are pointless now), 4) tests that run over some dtypes (these are fine).

This isn't the worse, but it goes from something that was probably thought out to something that wasn't.